portfolio/src/main/webapp/script.js

```diff
@@ -61,26 +66,27 @@ function addRandomFact() {
 function showBlogPost(postNum) {
   // Find the correct triangle to expand and switch its image.
   // Also change the onclick function for this button to hide the post.
-  const buttonId = "expandTriangle" + postNum;
-  let button = document.getElementById(buttonId);
+  const buttonId = 'expandTriangle' + postNum;
+  const button = document.getElementById(buttonId);
   console.log(button);
   if (button !== null) {
-    button.src = "/images/downtriangle.jpeg";
+    button.src = '/images/downtriangle.jpeg';
     button.onclick = function() {
       hideBlogPost(postNum);
     };
   }
 
   // Add blog post content to the correct area.
-  const blogPostId = "blogPostArea" + postNum;
-  let postArea = document.getElementById(blogPostId);
+  const blogPostId = 'blogPostArea' + postNum;
+  const postArea = document.getElementById(blogPostId);
   if (postArea !== null) {
     if (postNum > 0 && postNum <= posts.length) {
-      const elements = posts[postNum-1]
-      for (i = 0; i < elements.length; i++) {
-        const element = elements[i]
+      const elements = posts[postNum - 1];
+      let i = 0;
+      for (i; i < elements.length; i++) {
+        const element = elements[i];
         if (element !== null) {
-           postArea.appendChild(element);
+          postArea.appendChild(element);
```

Comment:
I'm a bit confused by this. How come you're modeling "one post" as a bunch of elements? Shouldn't there be a single element that represents "the post"? Then, you wouldn't need to iterate through elements and create them; just do it for the one higher level post element.